gamercade_console/src/console/contexts/draw_context.rs

```diff
@@ -287,13 +287,26 @@ impl DrawContext {
     }
 
     fn set_pixel_safe(&mut self, x: XCord, y: YCord, color: Color) {
-        let pixel_index = self.x_y_cord_to_pixel_buffer_index(x, y);
+        let pixel_index = match self.x_y_cord_to_pixel_buffer_index(x, y) {
+            Ok(v) => v,
+            _ => return,
+        };
         let color = color.into_pixel_data();
-        self.frame_buffer[pixel_index..pixel_index + BYTES_PER_PIXEL].copy_from_slice(&color);
+        let i = match pixel_index.checked_add(BYTES_PER_PIXEL) {
```

Comment:
Could `i` be renamed to something like... `pixel_bytes_range` or something more descriptive?

Reply:
Actually this code seems kinda hacky, I'll try to improve it to a more idiomatic one (or simply rename the variable :stuck_out_tongue_winking_eye: )